hack/update-codegen.sh

```diff
@@ -29,3 +29,11 @@ function docker_run() {
 }
 
 docker_run hack/update-codegen-dockerized.sh
+
+echo "=== Start resetting changes introduced by YEAR ==="
```

Comment:
is this supposed to only change the year for files which have actually been modified? How reliable is this, does it work if I have already committed some files?

it may be a bit overkill. IMO, it should be fine to update the year for all the files every time we run `./hack/update-codegen.sh`, even for files whose actual source has not been touched.

Reply:
actually I'm fine with this code, but why not put it in the "main" script (hack/update-codegen-dockerized.sh).

Reply:
I think it is better not to update every file since it is not hard to implement.

Moved code to update-codegen-dockerized.sh.